src/soilbiogeochem/SoilBiogeochemDecompCascadeBGCMod.F90

```diff
@@ -959,6 +946,18 @@ subroutine decomp_rate_constants_bgc(bounds, num_soilc, filter_soilc, &
             end if
          end do
       end do
+      pathfrac_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_l1s1) = 1.0_r8
+      pathfrac_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_l2s1)= 1.0_r8
+      pathfrac_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_l3s2) = 1.0_r8
+      pathfrac_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_s1s2) = f_s1s2(bounds%begc:bounds%endc,1:nlevdecomp)
+      pathfrac_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_s1s3) = f_s1s3(bounds%begc:bounds%endc,1:nlevdecomp)
+      pathfrac_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_s2s1) = f_s2s1
+      pathfrac_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_s2s3) = f_s2s3
+      pathfrac_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_s3s1) = 1.0_r8
+      if (.not. use_fates) then
+         pathfrac_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_cwdl2) = cwd_fcel
+         pathfrac_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_cwdl3) = cwd_flig
+      end if
```

Comment:
I now calc. `pathfrac_decomp_cascade` right after `decomp_k` in subr. decomp_rate_constants_bgc (and deprecated subr. decomp_rate_constants_cn) because otherwise the order of initializations did not permit adding `pathfrac_decomp_cascade` to history. In particular, leaving `pathfrac_decomp_cascade` in subr. init_decompcascade_bgc (i.e. unchanged):
1) as a `soilbiogeochem_state_inst` variable (i.e. unchanged) and adding to history resulted in messed up bounds for `cascade_donor_pool` in SoilBiogeochemStateType.F90
2) as a `soilbiogeochem_carbonflux_inst` variable and adding to history resulted in messed up bounds for `pathfrac_decomp_cascade` in subr. init_decompcascade_bgc
I'm happy to explain in greater detail if requested.

Reply:
I'll take your word on this :)